StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp11/SpacingRules/SA1010CSharp11UnitTests.cs

```diff
@@ -5,9 +5,46 @@
 
 namespace StyleCop.Analyzers.Test.CSharp11.SpacingRules
 {
+    using System.Threading.Tasks;
     using StyleCop.Analyzers.Test.CSharp10.SpacingRules;
+    using StyleCop.Analyzers.Test.Verifiers;
+    using Xunit;
+
+    using static StyleCop.Analyzers.Test.Verifiers.StyleCopCodeFixVerifier<
+        StyleCop.Analyzers.SpacingRules.SA1010OpeningSquareBracketsMustBeSpacedCorrectly,
+        StyleCop.Analyzers.SpacingRules.TokenSpacingCodeFixProvider>;
 
     public class SA1010CSharp11UnitTests : SA1010CSharp10UnitTests
     {
+        [Fact]
+        [WorkItem(3503, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3503")]
+        public async Task TestListPatternAsync()
+        {
+            var testCode = @"
+using System.Collections.Generic;
+
+namespace TestNamespace
+{
+    public class TestClass
+    {
+        public void TestMethod(List<int> x)
+        {
+            _ = x is [1];
+            _ = x is not [1];
+            _ = x is ([1] or [2]);
+            _ = x is ([1] or not [2]);
+            _ = x is ([1] and [1]);
+            _ = x is ([1] and not [2]);
```

Comment:
💡 This should be converted to a code fix test, which includes examples of various incorrect formatting that is then corrected (there is no need to separately test valid syntax, since all code fix tests automatically verify that the fixed code contains no undeclared diagnostics).

Reply:
This specific analyzer only suggests to remove spaces, never to add them. I assume there is a reason for that, even though I haven't figured it out yet :-)

Reply:
Interesting. Based on other analyzers (e.g. `{`), SA1010 would be the analyzer that wants to add the space here.

Reply:
Since this seems to be in line with the current implementation, would it be ok to merge this if I create another issue to investigate the resposibility of this analyzer?

Reply:
💭 Is this analyzer not responsible for reporting a this?

```
x in ( [1]);
       ^
```

In #3511 a very similar scenario does report a diagnostic.

Reply:
*Should* is a difficult word :-) The only diagnostic I would **like** to see in that example is SA1008. I am thinking that the list pattern's opening bracket itself should not trigger any opinion about preceding space, anymore than a literal value does. It is always something else that is causing the opinion, like a preceding open parenthesis, operator, keyword, etc. Before this PR, your example would trigger both SA1008 and SA1010. Is that a behaviour that you would like to keep? I could very well not be seeing the whole picture here, but I feel that not having an opinion at all is the correct behaviour for this rule in this specific case.

Reply:
I actually feel the same way about the property pattern changes made in SA1012, now that I think of it. I assume that both these statements will trigger two diagnostics:
```
_ = x is ( { P: 0 }, { P: 0 }) // SA1008 and SA1012
_ = x is [ { P: 0 }, { P: 0 }] // SA1010 and SA1012
```
With my thinking, only SA1008 and SA1010 should be triggered.

Reply:
Returning to SA1010 again, there was already a similar case implemented where this code only triggers SA1012 and not SA1010:
`var test = new Dictionary<ulong, string>() {[100] = "100" };`

Reply:
@sharwell?